Dockerfile

```diff
@@ -1,13 +1,22 @@
-FROM node:12.18.1-alpine
-RUN apk add --update\
- python \
- python3 \
- build-base \
- zeromq-dev \
- && rm -rf /var/cache/apk/*
+FROM ubuntu:20.04
+
+ENV NODE_VERSION=12.18.1
+ENV NVM_DIR=/usr/local/.nvm
+
+RUN mkdir -p ${NVM_DIR}
+RUN apt-get update && apt-get install -y curl
+
+RUN curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | bash
+RUN . "$NVM_DIR/nvm.sh" && nvm install ${NODE_VERSION}
+RUN . "$NVM_DIR/nvm.sh" && nvm use v${NODE_VERSION}
+RUN . "$NVM_DIR/nvm.sh" && nvm alias default v${NODE_VERSION}
+ENV PATH="${NVM_DIR}/versions/node/v${NODE_VERSION}/bin/:${PATH}"
+RUN node --version
+RUN npm --version
```

Comment:
not sure if you want it here at all, and if you do, can you move it into single `RUN` with npm?
```
RUN node --version
    && npm --version
    && npm install -g yarn
```

Reply:
Thanks @gimre-xymcity, already addressed that concern, it'll look like the following if we choose to upgrade it to `v12.x (latest)`. Any ideas on this => https://github.com/symbol/catapult-rest/pull/659#discussion_r707290619 ?
```
FROM ubuntu:20.04

RUN apt-get update && apt-get install -y curl \
    && curl -fsSL https://deb.nodesource.com/setup_12.x | bash - \
    && apt-get install -y nodejs \
    && node --version \
    && npm --version \
    && npm install -g yarn

WORKDIR /app
COPY . /app/catapult-rest
RUN cd catapult-rest && ./yarn_setup.sh
WORKDIR /app/catapult-rest/rest

```

Reply:
#perfect 👍

Reply:
yes, what you pasted in the comment looks good 👍

Reply:
just to make sure @Jaguar0625, the code pasted above will install the `nodejs v12.x (latest)` in each build(it won't be a fixed version). Version upgrades will be mostly security patches and npm upgrades but there is a risk of course.  I'm going to push the changes if we are all okay with using the latest version, 12.x.

Reply:
Okay, found a workaround to pin the version, please see the dockerfile content below.
```
FROM ubuntu:20.04

RUN apt-get update && apt-get install -y curl \
    && curl -o nodejs.deb https://deb.nodesource.com/node_12.x/pool/main/n/nodejs/nodejs_12.18.1-deb-1nodesource1_amd64.deb \
    && apt-get install -y ./nodejs.deb \
    && rm nodejs.deb \
    && rm -rf /var/lib/apt/lists/* \
    && node --version \
    && npm --version \
    && npm install -g yarn

WORKDIR /app
COPY . /app/catapult-rest
RUN cd catapult-rest && ./yarn_setup.sh
WORKDIR /app/catapult-rest/rest
```
Local tests worked fine. If everyone aggrees I'll push the changes using the fixed version?

Reply:
Nice finding. Fix version without nvm.

I'm ok using the old version although it's a good opportunity to upgrade the latest fix version 12.2.6

Reply:
i don't really see a compelling reason to use an older version? @0x6861746366574?

Reply:
Hey @0x6861746366574 , we need your input on this one, could you please share your view on this?
Current nodejs (in catapult-rest on the symbol network) version is `12.18.1`. 
We have 3 options:
1. Keep the version as it is as `12.18.1`. https://github.com/symbol/catapult-rest/pull/659#discussion_r710436996
2. Upgrade the version to `12.2.6` and pin it.
3. Use the latest `12.x` version on each build (currently the latest version is `12.22.6` but will probably change in the future, security patches, npm upgrades, etc.) https://github.com/symbol/catapult-rest/pull/659#discussion_r709508208

My vote would be on option 2 as well (with enough soak time in testnet ofc)

Reply:
**Option 3.** 

One of the biggest changes in cryptocurrency compared to traditional software development is that you want to make sure your dependencies are always operating on the latest version. The attack vectors for chains are very different compared to traditional financial systems - and the prize is far higher, too. 

There is additional burden on us and operators with frequent upgrades, yes, but, that's the entire point of network fees: ongoing reward for ongoing upkeep.

Reply:
Updated the dockerfile with the latest nodejs version as @0x6861746366574  suggested. 

Unfortunately, Travis is failing due to coveralls outage for some time now. We can wait until the service recovers or continue to merge.

> /home/travis/build/symbol/catapult-rest/catapult-sdk/node_modules/coveralls/bin/coveralls.js:19
      throw err;
      ^
Bad response: 503 <h2>This website is under heavy load (queue full)</h2><p>We're sorry, too many people are accessing this website at the same time. We're working on this problem. Please try again later.</p>

![image](https://user-images.githubusercontent.com/6256269/134066831-b86ded31-b119-436e-bbac-978285ac8b95.png)